app/models/action_page.rb

```diff
@@ -80,4 +81,8 @@ def image
   def no_drafts_on_homepage
     FeaturedActionPage.where(action_page_id: id).destroy_all unless published?
   end
+
+  def random_partner
+    partners.where(id: partners.pluck(:id).sample(1)).first
```

Comment:
`partners.sample` would have worked here, although style wise in that case `#random_partner` might be too simple to warrant its own method.